cclient/serialization/json/get_balances.c

```diff
@@ -56,29 +56,92 @@ retcode_t json_get_balances_serialize_request(
   return ret;
 }
 
-retcode_t json_get_balances_deserialize_response(serializer_t const *const s,
-                                                 char const *const obj,
-                                                 get_balances_res_t *out) {
+retcode_t json_get_balances_deserialize_request(serializer_t const *const s,
+                                                char const *const obj,
+                                                get_balances_req_t *const req) {
   retcode_t ret = RC_OK;
   cJSON *json_obj = cJSON_Parse(obj);
   cJSON *json_item = NULL;
   log_info(json_logger_id, "[%s:%d] %s\n", __func__, __LINE__, obj);
 
-  if (json_obj == NULL) {
-    log_error(json_logger_id, "[%s:%d] %s\n", __func__, __LINE__,
-              STR_CCLIENT_JSON_PARSE);
-    cJSON_Delete(json_obj);
-    return RC_CCLIENT_JSON_PARSE;
+  JSON_CHECK_ERROR(json_obj, json_item, json_logger_id);
+
+  ret = json_get_uint8(json_obj, "threshold", &req->threshold);
+  if (ret) {
+    goto end;
+  }
+
+  ret = json_array_to_hash243_queue(json_obj, "addresses", &req->addresses);
+  if (ret) {
+    goto end;
+  }
+
+  ret = json_array_to_hash243_queue(json_obj, "tips", &req->tips);
```

Comment:
`tips` is an optional parameter, no need `goto` here.

Reply:
This error code from `tips` element doesn't mean failed to `json_get_balances_deserialize_request`, `tips` is optional.